tests/test_downloader_handlers.py

```diff
@@ -239,7 +250,7 @@ def setUp(self):
         else:
             self.port = reactor.listenTCP(0, self.wrapper, interface=self.host)
         self.portno = self.port.getHost().port
-        self.download_handler = self.download_handler_cls(Settings())
+        self.download_handler = self.download_handler_cls.from_crawler(get_crawler())
```

Comment:
What do you think about using create_instance here, to simulate real usage better?